eZ/Publish/Core/Repository/ContentComparisonService.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Repository;
+
+use eZ\Publish\API\Repository\ContentComparisonService as ContentComparisonInterface;
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\Values\Content\VersionDiff\FieldValueDiff;
+use eZ\Publish\API\Repository\Values\Content\VersionDiff\FieldType\NoDiffResult;
+use eZ\Publish\API\Repository\Values\Content\VersionDiff\VersionDiff;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Base\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\Comparison\ComparisonEngineRegistry;
+use eZ\Publish\Core\Comparison\FieldRegistry;
+use eZ\Publish\Core\Repository\Helper\ContentTypeDomainMapper;
+use eZ\Publish\SPI\Persistence\Content;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\Handler as ContentHandler;
+use eZ\Publish\SPI\Persistence\Content\Type;
+use eZ\Publish\SPI\Persistence\Content\Type\Handler as ContentTypeHandler;
+
+class ContentComparisonService implements ContentComparisonInterface
+{
+    /** @var \eZ\Publish\SPI\Persistence\Content\Handler */
+    private $contentHandler;
+
+    /** @var \eZ\Publish\SPI\Persistence\Content\Type\Handler */
+    private $contentTypeHandler;
+
+    /** @var \eZ\Publish\Core\Comparison\FieldRegistry */
+    private $fieldRegistry;
+
+    /** @var \eZ\Publish\Core\Comparison\ComparisonEngineRegistry */
+    private $comparatorEngineRegistry;
+
+    /** @var \eZ\Publish\Core\Repository\Helper\ContentTypeDomainMapper */
+    private $contentTypeDomainMapper;
+
+    /** @var \eZ\Publish\API\Repository\PermissionResolver */
+    private $permissionResolver;
+
+    public function __construct(
+        ContentHandler $contentHandler,
+        ContentTypeHandler $contentTypeHandler,
+        FieldRegistry $fieldRegistry,
+        ComparisonEngineRegistry $comparatorEngineRegistry,
```

Comment:
inconsistent naming:
```suggestion
        ComparisonEngineRegistry $comparisonEngineRegistry,
```